internal/pipe/docker/api_kaniko.go

```diff
@@ -0,0 +1,82 @@
+package docker
+
+import (
+	"fmt"
+	"github.com/goreleaser/goreleaser/pkg/config"
+	"os"
+
+	"github.com/goreleaser/goreleaser/pkg/context"
+)
+
+func init() {
+	registerImager(useKaniko, kanikoImager{})
+}
+
+const (
+	kanikoExecutorImage = "gcr.io/kaniko-project/executor:latest"
+	kanikoWorkspace     = "/workspace"
+)
+
+type kanikoImager struct{}
+
+// Push is a no-op and it returns nil because kaniko performs build and push in a single step.
+// https://github.com/GoogleContainerTools/kaniko#how-does-kaniko-work
+func (i kanikoImager) Push(_ *context.Context, _ string, _ []string) error {
+	return nil
+}
+
+func (i kanikoImager) Build(ctx *context.Context, docker config.Docker, root string, images, flags []string) error {
```

Comment:
We need to pass `config.Docker` here so that the configuration values can be used to construct the build command.